crates/subspace-networking/src/node_runner.rs

```diff
@@ -206,9 +212,7 @@ impl NodeRunner {
         }
 
         // Maintain minimum connected peers number.
-        if connected_peers.len() < CONNECTED_PEERS_THRESHOLD
-            && connected_peers.len() < self.max_established_total as usize
-        {
+        if connected_peers.len() < self.max_established_outgoing_connections as usize {
```

Comment:
Should we account for pending outgoing connections here? Otherwise I think we can overshoot quite far under certain circumstances.